### PR TITLE
Switch most of container host tags to host metadata

### PR DESCRIPTION
**Pull request by hkaj**
*Note: Please remember to review the Datadog [Contribution Guidelines](https://github.com/DataDog/dd-agent/blob/master/CONTRIBUTING.md)
if you have not yet done so.*

### What does this PR do?

Move all host tags coming from docker and orchestrators to host metadata with the exception of node labels in k8s because they're legit.

### Motivation

This metadata is not really useful as tags and it's mostly for analysis purpose. Therefore metadata is a better fit that tags - especially because host metadata doesn't churn contexts when updated.

### Testing Guidelines

An overview on [testing](https://github.com/DataDog/dd-agent/blob/master/tests/README.md)
is available in our contribution guidelines.

### Additional Notes

This will require some work in the app to display it. Still figuring that out.

### CODE REVIEW

File: utils/orchestrator/baseutil.py [hunk 1]
```diff
@@ -22,6 +22,8 @@ class BaseUtil:
       - _get_transient_tags: tags can change and won't be cached (TODO)
       - invalidate_cache: custom cache invalidation logic
       - is_detected (staticmethod)
+      - get_host_tags: list of tags that are applied to the host in Datadog
+      - get_host_metadata: dict of container-related host metadata (typically node labels)
```
Comment (xvello) [inline utils/orchestrator/baseutil.py]: Bad example, aren't we exporting node labels as tags and not metadata?

File: utils/kubernetes/kubeutil.py [hunk 2]
```diff
@@ -526,10 +548,6 @@ def extract_event_tags(self, event):
             tags.append('node_name:%s' % event['source']['host'])
         if 'kind' in event.get('involvedObject', {}):
             tags.append('object_type:%s' % event['involvedObject'].get('kind', '').lower())
-        if 'name' in event.get('involvedObject', {}):
```
Comment (xvello) [inline utils/kubernetes/kubeutil.py]: Probably a bad rebase (these are quite new), please keep them

File: utils/orchestrator/baseutil.py (hunk 1 above)
Comment (hkaj) [reply utils/orchestrator/baseutil.py]: oops yeah wrong line

**Comment by hkaj** [comment]
@xvello can you have a second look 🙏

**Pull request merged**